src/custom/utils/metadata.ts

```diff
@@ -43,3 +47,14 @@ export function generateAppDataDoc(metadata: MetadataDoc = {}): AppDataDoc {
     },
   }
 }
+
+export async function uploadMetadataDocToIpfs(appDataDoc: AppDataDoc): Promise<string> {
+  try {
+    const cid = await uploadTextFileToIpfs(appDataDoc)
+    await pinByHash(cid)
+    const { digest } = multihashes.decode(new CID(cid).multihash)
+    return `0x${Buffer.from(digest).toString('hex')}`
+  } catch (err) {
+    throw err.message ? err : new Error(err)
```

Comment:
I mentioned this also when u shared by chat. What's the goal of this error handling? 
why not just do nothing if you just re-throw